### PR TITLE
Add conversion rule from pointer to const pointer to Dyno.

### PR DESCRIPTION
**Pull request by DanilaFe**
This was the only missing conversion in Dyno -- non-const pointers can be safely converted to const ones.

This PR also tweaks the instantiation rules to allow `c_ptrConst(?t)` to be instantiated by `c_ptr(X)` etc.. This isn't present in production, but it doesn't seem explicitly ruled out -- the case just didn't seem considered.

Reviewed by @arezaii -- thanks!

## Testing
- [x] dyno test suite
- [x] paratest

### CODE REVIEW

File: frontend/lib/resolution/can-pass.cpp [hunk 1]
```diff
@@ -333,8 +333,14 @@ bool
...
                               const Type* actualT,
                               const Type* formalT) {
-  if (actualT->isCPtrType()) {
+  if (auto actualPtr = actualT->toCPtrType()) {
     if (auto formalPtr = formalT->toCPtrType()) {
+      // Allow constness casts: an int* can be a const int*.
+      // In Chapel lingo, `c_ptr(int)` is passable to `c_ptrConst(int)`.
+      if (formalPtr->isConst() && !actualPtr->isConst() &&
+          formalPtr->eltType() == actualPtr->eltType())
+        return true;
```
Comment (arezaii) [inline frontend/lib/resolution/can-pass.cpp:342]: what if the eltTypes are not exactly the same, but can be passed? For example, I was expecting a `c_ptr(int(32))` actual to pass to a formal defined as `c_ptrConst(int(?w))`

**Review by arezaii — COMMENTED** [review]
I was not sure about the more complex cases, where the `eltType` is not exactly the same but can be treated as passable. What do you think?

File: frontend/test/resolution/testCPtr.cpp [hunk 2]
```diff
@@ -245,6 +245,52 @@ static void test17() {
   });
 }
 
+static void test18() {
+  testCPtrArg("c_ptrConst(int)", "c_ptr(int)", [](const TypedFnSignature* fn, const CPtrType* t, ErrorGuard& eg) {
+    assert(t);
```
Comment (arezaii) [inline frontend/test/resolution/testCPtr.cpp:250]: should this test also assert that the `TypedFnSignature` is found?

Comment (DanilaFe) [reply frontend/test/resolution/testCPtr.cpp:250]: I think `formalTypePtr` couldn't be non-null if `fn` is null, but I'll add the check anyway.

File: frontend/lib/resolution/can-pass.cpp (hunk 1 above)
Comment (DanilaFe) [reply frontend/lib/resolution/can-pass.cpp:342]: For instantiation, this is a good point -- I've ensured this works.

I don't think it should work for conversions (c_ptrConst(int(64)) <- c_ptr(int(32))), though.

**Review by arezaii — APPROVED** [review]
all looks good, thank you!

**Pull request merged**